### PR TITLE
Add phoenix query runner.

### PR DESCRIPTION
**Pull request by combineads**
This is an apache phoenix query runner. The query runner presently does not support Kerberos authentication. But I plan to implement it soon.
screenshot a query runner configuration

### CODE REVIEW

**Comment by combineads** [comment]
@arikfr May I ask you for a review?

File: redash/query_runner/phoenix.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            json_data = json_dumps(data)
+            error = None
+            cursor.close()
+        except (KeyboardInterrupt, InterruptException) as e:
+            error = "Query cancelled by user."
+            json_data = None
+        except Exception as ex:
+            json_data = None
+            error = 'error.{}.{}'.format(query, ex)
+            if not isinstance(error, basestring):
```
Comment (arikfr) [inline redash/query_runner/phoenix.py]: Why error want be a `basestring`?

File: redash/query_runner/phoenix.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            data = {'columns': columns, 'rows': rows}
+            json_data = json_dumps(data)
+            error = None
+            cursor.close()
+        except (KeyboardInterrupt, InterruptException) as e:
+            error = "Query cancelled by user."
+            json_data = None
+        except Exception as ex:
+            json_data = None
+            error = 'error.{}.{}'.format(query, ex)
```
Comment (arikfr) [inline redash/query_runner/phoenix.py]: The error message is shown to the user and should be (to some degree) user friendly. Not sure this qualifies :)

Maybe returning just the exception message is enough?

**Comment by combineads** [comment]
@arikfr Thank you for your review.
I improved error handling.

File: redash/query_runner/phoenix.py (hunk 2 above)
Comment (combineads) [reply redash/query_runner/phoenix.py]: Error handling is sufficient. Users can identify errors with code and message. Please refer to https://python-phoenixdb.readthedocs.io/en/latest/api.html#phoenixdb.Error.

**Comment by arikfr** [comment]
Thanks!

**Pull request merged**